crates/fuel-core/src/service.rs

```diff
@@ -89,6 +90,8 @@ pub struct SharedState {
     pub executor: ExecutorAdapter,
     /// The config of the service.
     pub config: Config,
+    /// notifier to trigger block production
+    pub block_production_trigger: BlockProductionTrigger,
```

Comment:
We don't need it to be a public field. Even more, we don't need to store it at all, you can pass it via `TaskParams` of the `FuelService` and trigger it there

Reply:
addressed in fda6b1e2bc